src/features/vault.ts

```diff
@@ -92,8 +92,11 @@ export const toggleEncrypt = async (): Promise<void> => {
         editBuilder.replace(
           selection,
           encryptedText.replace(
-            /\n/g,
-            `\n${' '.repeat(selection.start.character)}`
+            /\n\s*/g,
+            `\n${' '.repeat(
+              (getIndentationLevel(editor, selection) + 1) *
+                Number(editor.options.tabSize)
+            )}`
```

Comment:
This is quite complex. Maybe move it out and assign it to a var?

Reply:
Done.